### PR TITLE
Prune kubernetes-maintainers github group

### PR DESCRIPTION
**Pull request by dims**
`kubernetes-maintainers` have direct write access to kubernetes/kubernetes repository, so we need to be vigilant as to who has access and what they need it for. So to start with, let's make sure we drop folks who are not really active anymore.

- activity : [Developer Activity Counts for 1 year](https://k8s.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%20year&var-metric=contributions&var-repogroup_name=All&var-repo_name=kubernetes%2Fkubernetes&var-country_name=All&var-companies=All)
- leads : one of leads@k8s.io group (Yes/No/NA). NA means they have activity so it does not matter if they are lead or not.

| github id | activity | lead   |
| --------  | -------- | -------- |
| apelisse    |  2123  |  NA  |
| BenTheElder    |  4709  |  NA  |
| bgrant0607    |  0  |  No  |
| brendandburns    |  1213  |  NA  |
| bsalamat    |  0  |  No  |
| caesarxuchao    |  0  |  No  |
| cheftako    |  516  |  NA  |
| childsb    |  0  | No  |
| cjcullen    |  0  | No  |
| davidopp    |  0  | No   |
| dchen1107    |  260  | NA   |
| deads2k    |  1252  |  NA  |
| derekwaynecarr    |  243  |  NA  |
| dims    |  2422  |  NA  |
| eparis    |  0  |  No  |
| erictune    |  0  | No   |
| feiskyer    |  901  |  NA  |
| fejta    |  99  | NA   |
| freehan    |  0  |  No  |
| grodrigues3    |  0  |  No  |
| ingvagabund    |  707  |  NA  |
| ixdy    |  0  |  No  |
| janetkuo    |  41  |  NA  |
| jbeda    |  0  |  No  |
| jessfraz    |  0  |  No  |
| jingxu97    |  413  |  NA  |
| jsafrane    |  1329  | NA   |
| justinsb    |  1701  | NA   |
| k82cn    |  56  |  NA  |
| kevin-wangzefeng    |  0  | No  |
| krousey    |  0  | No   |
| lavalamp    |  2668  |  NA  |
| liggitt    |  8757  |  NA  |
| luxas    |  0  |  No  |
| mikedanese    |  222  | NA   |
| mml    |   0 | No   |
| mtaufen    |  69  |  NA  |
| mwielgus    | 463   |  NA  |
| ncdc    |  250  |  NA  |
| piosz    |  0  |  No  |
| pmorie    | 0   |  No  |
| pwittrock    |  95  |  NA  |
| quinton-hoole    |  0  |  No  |
| Random-Liu    |  0  |  No  |
| saad-ali    |  0  |  Yes  |
| smarterclayton    | 1169   |  NA  |
| soltysh    |  1409  | NA   |
| spiffxp    |  295  |  NA  |
| sttts    |  145  |  NA  |
| tallclair    |  1425  | NA   |
| thockin    |  4415  |  NA  |
| timothysc    |  0  |  No  |
| vishh    |  0  |  No  |
| wojtek-t    | 3194   | NA   |
| xiang90    |  0  |  No  |
| yifan-gu    | 0   | No   |
| yujuhong    | 0   | No   |
| zmerlynn    | 0   | No   |

### CODE REVIEW

**Comment by dims** [comment]
cc @bgrant0607 @bsalamat @caesarxuchao @childsb @cjcullen @davidopp @eparis @erictune @freehan @grodrigues3 @ixdy @janetkuo @jbeda @jessfraz @kevin @krousey @luxas @mml @piosz @pmorie @pwittrock @quinton @Random @timothysc @vishh @xiang90 @yifan @yujuhong @zmerlynn

**Comment by dims** [comment]
Trying to contact the folks via:
- github mention (see above)
- emailed their last known email address i could find

**Comment by enj** [comment]
LGTM

**Comment by liggitt** [comment]
one data point in the table that stood out as inaccurate is @janetkuo, who had 41 contributions in the last year and is a sig-apps lead

**Comment by dims** [comment]
> one data point in the table that stood out as inaccurate is @janetkuo, who had 41 contributions in the last year and is a sig-apps lead

Apologies. totally typoed! sorry @janetkuo

**Comment by mrbobbytables** [comment]
Overall LGTM, was recently talking about doing something similar - will wait for others to ack

**Comment by BenTheElder** [comment]
Thanks @dims ! Bob and I were just discussing the need to do this yesterday :-)

I think this list checks out now https://github.com/kubernetes/org/pull/4139#issuecomment-1491887371, let's get some more eyes on to see if we missed anything else 😅 

-----

For folks following along, I know dims knows this but:

For anyone unfamiliar with the project we have [branch protections](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches) automatically [enforced](https://docs.prow.k8s.io/docs/components/optional/branchprotector/) that require merging be by the merge robot following our configured merge policies (passing tests, approve+lgtm, ...). Write permission to the repo however allows you to edit PR/issue metadata and could be abused for example to convince the robot to consider a PR for the merge queue without lgtm from another contributor (which is at least rather rate limited by the need to still pass our lengthy tests at the latest of both the PR branch and the repo branch before merge).

So, only the admin permission group (MUCH smaller, the team that administers our Github organizations) has "force merge / force push" to Kubernetes release branches etc., but this group absolutely needs pruning and it's possible to do other disruptive things with this permission. This permission is typically legitimately used by project maintainers to do things like updating issue checklists that the, other, more complex webhook `/command`-comment based bot either doesn't support yet or isn't working correctly (`/release-note-edit` comes to mind recently).

**TLDR**: This group needs to be kept pruned and these permissions are abusable but maybe not in the way "direct write access" might sound like, generally speaking they cannot push/force-merge.

**Comment by bgrant0607** [comment]
Thanks for the ping. I can be removed.

**Comment by bsalamat** [comment]
I had been away from the K8s world for a few years, but I am coming back. That said, feel free to remove me from the maintainers for now.

**Comment by dims** [comment]
> Thanks for the ping. I can be removed.

thanks @bgrant0607 !

**Comment by dims** [comment]
> That said, feel free to remove me from the maintainers for now.

Thanks @bsalamat Also welcome back!

**Comment by zmerlynn** [comment]
I haven't been active in k/k for a few years and can be removed.

File: config/kubernetes/org.yaml [hunk 1]
```diff
@@ -2000,62 +2000,34 @@ teams:
...
     - derekwaynecarr
     - dims
-    - eparis
-    - erictune
     - feiskyer
     - fejta
-    - freehan
-    - grodrigues3
     - ingvagabund
-    - ixdy
```
Comment (ixdy) [inline config/kubernetes/org.yaml:2022]: LGTM

**Comment by cblecker** [comment]
LGTM

**Comment by jbeda** [comment]
LGTM 😢

I have aspirations one day of getting involved again and starting from the bottom :)

**Comment by dims** [comment]
> I have aspirations one day of getting involved again and starting from the bottom :)

💐🌹🌼 !!

**Comment by cjcullen** [comment]
LGTM. I don't need direct write access.

**Comment by piosz** [comment]
LGTM 😢

**Comment by quinton-hoole** [comment]
Yup, I've not contributed for years - happy to be removed.  Like Joe, I have aspirations of chipping in again some time.

**Comment by xiang90** [comment]
LGTM.

**Comment by jessfraz** [comment]
Thank you! Happy to have less notifications since I'm not helping! <3 you all, will see you at some nerd event in the future I'm sure :)

**Comment by dims** [comment]
Folks, How about i leave this open until end of this week and merge it either friday evening (7th) or monday morning (10th) ... if anyone objects even after this gets merged, we can always do a partial rollback.

Big thanks to everyone one of you who have played a pivotal role(s) in our community!

**Comment by erictune** [comment]
Ack.  I don't need this access anymore.

**Comment by erictune** [comment]
Are you sure about @freehan?  He is the Sig-instrumentation chair.

**Comment by cblecker** [comment]
@erictune Did you mean @logicalhan ?

**Comment by erictune** [comment]
Yeah, sorry.

**Comment by dims** [comment]
/hold cancel

Thanks to you all!

image
